### PR TITLE
Support Modern Solaris 

### LINKED ISSUE #1251: Support Modern Solaris
In Issue #187 we removed the support for (ancient) Solaris. However, it doesn't seem like it was quite as dead as originally anticipated. Still our support was badly bitrotten, busted, and indeed quite the mess.

This, though, is the meta/tracking issue that will bring in code changes and build system support for modern Solaris and associated open forks (Illumos/OpenIndiana).

----

**Work Repository:** https://github.com/athenian200/UXP
**Compare:** https://github.com/MoonchildProductions/UXP/compare/master...athenian200:master
**Discussion Thread:** https://forum.palemoon.org/viewtopic.php?f=5&t=22899

**Assigned:** @athenian200
**Contact:** @mattatobin

### PR DESCRIPTION
**Pull request by athenian200**
Resolves #1251.

It looks daunting, I know, because it touches many parts of the tree, but I documented everything in the commit comments and even though it touches several parts, it isn't a huge number of changes to each part. It touches the whole thing in shallow ways rather than extensively changing one thing in terms of functionality.

If I did this incorrectly, I should probably note I've never done a pull request before, so... there's that. It may not be up to the usual contribution standards because I'm somewhat new to working with git and GitHub, and I apologize in advance if things look a bit... rough.

### CODE REVIEW

File: ipc/chromium/src/base/sys_info_posix.cc [hunk 1]
```diff
@@ -133,6 +133,10 @@ std::string SysInfo::OperatingSystemName() {
   return std::string(info.sysname);
 }
 
+// Solaris <sys/utsname.h> contains "extern struct utsname utsname;" 
+// As a consequence, any use of utsname has to be proceeded with struct on
```
Comment (wolfbeast) [inline ipc/chromium/src/base/sys_info_posix.cc]: Nit: s/proceeded/preceded/

File: toolkit/library/libxul.mk [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 EXTRA_DEPS += $(topsrcdir)/toolkit/library/libxul.mk
 
-ifeq (Linux,$(OS_ARCH))
+ifeq (Linux,$(OS_ARCH))
```
Comment (wolfbeast) [inline toolkit/library/libxul.mk]: Nit: trailing whitespace

File: xpcom/reflect/xptcall/md/unix/moz.build [hunk 3]
```diff
@@ -56,7 +56,7 @@ if CONFIG['OS_ARCH'] == 'SunOS':
                 'xptcinvoke_x86_64_unix.cpp',
                 'xptcstubs_x86_64_linux.cpp'
             ]
-    elif CONFIG['OS_TEST'] == 'x86':
+    elif '86' in CONFIG['OS_TEST']:
```
Comment (wolfbeast) [inline xpcom/reflect/xptcall/md/unix/moz.build]: ... Why was this incorrect?

**Comment by athenian200** [comment]
Okay, I fixed everything except the OS_TEST thing, because that's due to an issue with how OS_TEST works.

**Pull request merged**